src/Controller/OpenidController.php

```diff
@@ -0,0 +1,159 @@
+use Psr\Http\Message\ServerRequestInterface;
+
+class OpenidController extends AbstractController
+{    
+    private $keys;
+	private $openIdConfiguration;
+	private $authServerConfig;
+	private $authServerFactory;
+	
+    public function __construct(){
+        // parent::__construct();
+        require_once(__DIR__.'/../../vendor/autoload.php');
+
+        $this->keys = $this->getKeys();
+		$this->openIdConfiguration = $this->getOpenIdConfiguration();
+		
+		$this->authServerConfig = $this->createConfig();
+		$this->authServerFactory = (new \Pdsinterop\Solid\Auth\Factory\AuthorizationServerFactory($this->authServerConfig))->create();
+    }
+    private function linkToRoute($route) {
+        return '/some/route';
+    }
+    private function getBaseUrl() {
+        return 'http://localhost/';
+    }
+    private function getAbsoluteUrl($relativeUrl) {
+        return 'http://localhost/some/route';
+    }
+    private function getOpenIdConfiguration() {
+		return array(
+			'issuer' => $this->getBaseUrl(),
+			'authorization_endpoint' => $this->getAbsoluteUrl($this->linkToRoute("solid.server.authorize")),
+			'jwks_uri' => $this->getAbsoluteUrl($this->linkToRoute("solid.server.jwks")),
+			"response_types_supported" => array("code","code token","code id_token","id_token code","id_token","id_token token","code id_token token","none"),
+			"token_types_supported" => array("legacyPop","dpop"),
+			"response_modes_supported" => array("query","fragment"),
+			"grant_types_supported" => array("authorization_code","implicit","refresh_token","client_credentials"),
+			"subject_types_supported" => ["public"],
+			"id_token_signing_alg_values_supported" => ["RS256"],
+			"token_endpoint_auth_methods_supported" => "client_secret_basic",
+			"token_endpoint_auth_signing_alg_values_supported" => ["RS256"],
+			"display_values_supported" => [],
+			"claim_types_supported" => ["normal"],
+			"claims_supported" => [],
+			"claims_parameter_supported" => false,
+			"request_parameter_supported" => true,
+			"request_uri_parameter_supported" => false,
+			"require_request_uri_registration" => false,
+			"check_session_iframe" => $this->getAbsoluteUrl($this->linkToRoute("solid.server.session")),
+			"end_session_endpoint" => $this->getAbsoluteUrl($this->linkToRoute("solid.server.logout")),
+			"token_endpoint" => $this->getAbsoluteUrl($this->linkToRoute("solid.server.token")),
+			"userinfo_endpoint" => $this->getAbsoluteUrl($this->linkToRoute("solid.server.userinfo")),
+			"registration_endpoint" => $this->getAbsoluteUrl($this->linkToRoute("solid.server.register")),
+	//		"sharing_endpoint" => $this->getAbsoluteUrl($this->linkToRoute("solid.server.sharing"))
+		);
+    }
+    private function getKeys() {
+		// FIXME: read these from the solid config in nextcloud;
+		$encryptionKey = 'P76gcBVeXsVzrHiYp4IIwore5rQz4cotdZ2j9GV5V04=';
+		$privateKey = <<<EOF
+-----BEGIN RSA PRIVATE KEY-----
+MIIEpAIBAAKCAQEAvqb0htUFZaZ+z5rn7cHWg0VzsSoVnusbtJvwWtHfD0T0s6Hb
```

Comment:
Is it bad that I have seen this key so often that I can reckognize it? :joy_cat: